Gemfile

```diff
@@ -54,6 +54,7 @@ group :test do
   gem "capybara-screenshot"
   gem "database_cleaner-active_record", "~> 1.8.0"
   gem "rake"
+  gem "rails-controller-testing"
```

Comment:
I'm not super thrilled with adding this extra gem but it's fairly maintained and easy to remove later if it gets obsolete :)
https://github.com/rails/rails-controller-testing
> This gem brings back assigns to your controller tests as well as assert_template to both controller and integration tests.
> Note: This gem is only useful once assigns and assert_template have been removed from Rails.

Reply:
@compwron 

Thanks for the review. Perhaps we might consider removing the controller test. Since writing the tests, I've taken a little time to try to learn when and why the ability to test with `assigns` was removed. The removal of the ability to test instance variables was because:

> Testing what instance variables are set by your controller is a bad idea. That's grossly overstepping the boundaries of what the test should know about. You can test what cookies are set, what HTTP code is returned, how the view looks, or what mutations happened to the DB, but testing the innards of the controller is just not a good idea.

See this Rails PR that removed it: https://github.com/rails/rails/issues/18950

If there is an alternative way to test the internal logic of the controller, I'm open to that.

Please let me know if the team thinks the controller tests here should be removed.

Reply:
We can change the a little and remove the gem if we want to. Keep the tests for now. :)

Reply:
We are using all request specs now, and no controller tests. It would be great if this spec could get rewritten as a request spec.

Reply:
@littleforest and @compwron 

Thank you for the information. 

I have a couple of things to ask about and note:

1. We currently have a system test for this change at https://github.com/rubyforgood/casa/blob/main/spec/system/case_contacts/new_spec.rb#L330-L362. I think this gets us what we want for testing this change (and I'm not sure the controller test a lot of value compared to that). Perhaps these tests cover what we really care about.
2. When looking through the request tests, what I see is verification of interactions in the request. I don't see any tests in the current requests test suite that verify what data are available to the user. Is there an example somewhere of testing the kind of logic that has changed here? Specifically, here that would be making sure that only certain contact types are available as choices for the user. Is there something similar that would be done in a requests test suite?

After the conversation here and looking over the test suite in general, I'm inclined to believe that the system tests cover what we want. If there is a need to add testing, I'm happy to do so. Having a sample of something similar tested in the requests test suite would help with that.